src/config.rs

```diff
@@ -82,6 +82,50 @@ macro_rules! create_config {
             $(pub $i: Option<$ty>),+
         }
 
+        // This trait and the following impl blocks are there only so that we
+        // can use UCFS inside the get_docs() function on builtin types for configs.
+        trait IsConfigType {
+            fn get_variant_names() -> Vec<&'static str>;
```

Comment:
I agree that it is an ugly way of accomplishing printing the variant names; @nrc had mentioned in the issue that we wanted to print variant names, but if there is a better way, I would be more than happy to change this.